hangups/http_utils.py

```diff
@@ -5,57 +5,126 @@
 import collections
 import logging
 
-from hangups import exceptions
+from hangups import exceptions, channel
 
 logger = logging.getLogger(__name__)
 CONNECT_TIMEOUT = 30
 REQUEST_TIMEOUT = 30
 MAX_RETRIES = 3
 
-FetchResponse = collections.namedtuple('FetchResponse', ['code', 'body',
-                                                         'cookies'])
+FetchResponse = collections.namedtuple('FetchResponse', ['code', 'body'])
 
 
-@asyncio.coroutine
-def fetch(method, url, params=None, headers=None, cookies=None, data=None,
-          connector=None):
-    """Make an HTTP request.
+class ClientSession(aiohttp.ClientSession):
```

Comment:
The advantage is we could avoid depending on implementation details of `aiohttp.ClientSession` such as `_request`, and we could expose a more specialized interface to the rest of the application. Could we get by with just exposing the methods `fetch` and `close`?